src/services/web-monitoring-db.js

```diff
@@ -401,3 +402,28 @@ function parseDiff (data) {
   }
   return data;
 }
+
+/**
+ * Create a function that will throw an error when given parsed response data
+ * that contains error information. Use this to reject the promise for an HTTP
+ * request.
+ *
+ * @param {string} summary If there are multiple errors or an error message
+ *   can not be found in the response, use this as the error message.
+ * @returns {Function}
+ */
+function throwErrorResponse (summary) {
+  return data => {
+    if (data.errors) {
+      const firstMessage = data.errors[0].title || data.errors[0].message;
+      const message = data.errors.length > 1 && summary || firstMessage;
+      const error = new Error(message);
+      error.details = data.errors;
+      throw error;
+    }
+    else if (data.error) {
+      throw new Error(data.error.title || data.error.message || summary);
+    }
+    return data;
```

Comment:
Similarly, the default case (and more likely scenario) is returning `data` and not throwing an error, which was confusing for me at first as well. Not sure what a good solution is because I'm new to using promises and we've seen how I've mangled them in the past. Maybe rename to `checkErrorResponse`???